test_dependencies.md

```diff
@@ -1,21 +1,75 @@
 
-New versions of paramiko complain about being unable to import SSH2_AGENTC_REQUEST_IDENTITIES, 1.10.2 works.
+## Basics
 
-With the new auth2 clients and server we may no longer need rsa, paramiko, Crypt::OpenSSL::RSA, Convert::PEM
+```sh
+$ sudo apt-get update
+$ sudo apt-get install build-essential libfontconfig
+```
+
+## Java
 
-Make sure nose2 is using python2.
+```sh
+$ sudo apt-get remove openjdk*
+$ sudo apt-get autoclean
+$ sudo apt-get install openjdk-8-jdk ant
+```
 
-pip install requests nose2 jsonrpcbase rsa paramiko==1.10.2
+## Docker
 
-cpan Moose
+```sh
+$ sudo apt-get update
+$ sudo apt-get install apt-transport-https ca-certificates curl software-properties-common
+$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+$ sudo apt-key fingerprint 0EBFCD88
+$ sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+$ sudo apt-get update
+$ sudo apt-get install docker-ce
+```
 
-cpan -f -i RPC::Any
+## JS
 
-cpan Bio::KBase::Exceptions
+CasperJS, > 1.1.0. http://casperjs.org/ 
 
-cpan Parse::Yapp Devel::StackTrace Lingua::EN::Inflect Template File::Slurp Cwd JSON Data::UUID XML::Dumper JSON::RPC::Client Exception::Class Config::Simple Digest::SHA1 Crypt::OpenSSL::RSA Convert::PEM DateTime MIME::Base64 URI Object::Tiny::RW Plack File::ShareDir::Install YAML TAP::Harness Plack::Middleware::CrossOrigin RPC::Any::Server::JSONRPC::PSGI
+```sh
+$ git clone git://github.com/casperjs/casperjs.git`
+$ cd casperjs
+$ ln -sf `pwd`/bin/casperjs /usr/local/bin/casperjs
+```
 
-CasperJS must be installed, > 1.1.0. http://casperjs.org/ .  PhantomJS 1.9.x also required
+PhantomJS 1.9.x
+```sh
+$ export PHANTOM_JS="phantomjs-1.9.8-linux-x86_64"
+$ wget https://bitbucket.org/ariya/phantomjs/downloads/$PHANTOM_JS.tar.bz2
+$ tar xvjf $PHANTOM_JS.tar.bz2
+$ mv $PHANTOM_JS /opt/
+$ ln -sf /opt/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+```
+
+## Python
+
+```sh
+sudo apt-get install python python-setuptools
+easy_install pip
+pip install requests nose2 jsonrpcbase
+```
+
+New versions of paramiko complain about being unable to import SSH2_AGENTC_REQUEST_IDENTITIES, 1.10.2 works.
+
+## Perl
+
+```sh
+$ cpan  # run configuration
+$ sudo cpan Moose
+$ sudo cpan -f -i RPC::Any
+$ sudo cpan Bio::KBase::Exceptions
+$ sudo cpan Parse::Yapp Devel::StackTrace Lingua::EN::Inflect Template File::Slurp Cwd JSON Data::UUID XML::Dumper JSON::RPC::Client Exception::Class Config::Simple Digest::SHA1 Crypt::OpenSSL::RSA Convert::PEM DateTime MIME::Base64 URI Object::Tiny::RW Plack File::ShareDir::Install YAML TAP::Harness Plack::Middleware::CrossOrigin RPC::Any::Server::JSONRPC::PSGI
+```
+
+Must force RPC::Any, Moose updates cause XML tests to fail.
+
+## Misc
+
+With the new auth2 clients and server we may no longer need rsa, paramiko, Crypt::OpenSSL::RSA, Convert::PEM
```

Comment:
Not installing rsa/paramiko anymore, should we be?

Reply:
If the tests pass then Paramiko may be unneeded at this point. IIRC Paramiko was useful for some ssh-key based authentication features in the old auth2, but I don't think anyone ever used that

Reply:
Not sure what the perl clients still use though.